### PR TITLE
New plugins and improvements SAP-642

### PR DESCRIPTION
**Pull request by as114**
- Added sample to plate assignment plugin.
- Allow file re-upload for DDPCR result analysis plugin.
- Auto Index assignment for RNA samples.
- Generate sample-sheet for STR workflow.

### CODE REVIEW

File: src/main/java/com/velox/sloan/cmo/workflows/IgoLimsPluginUtils/AliquotingTags.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package com.velox.sloan.cmo.workflows.IgoLimsPluginUtils;
+
+public interface AliquotingTags {
+
+    public static final String ALIQUOT_STARTING_CONCENTRATION = ".*<!--\\s*ALIQUOT\\s*STARTING\\s*CONCENTRATION\\s*-->.*";
```
Comment (dmcmanam) [inline src/main/java/com/velox/sloan/cmo/workflows/IgoLimsPluginUtils/AliquotingTags.java]: 'public static final' redundant for interface, let's remove it.

File: src/main/java/com/velox/sloan/cmo/workflows/IgoLimsPluginUtils/Tags.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.velox.sloan.cmo.workflows.IgoLimsPluginUtils;
+
+public interface Tags {
```
Comment (dmcmanam) [inline src/main/java/com/velox/sloan/cmo/workflows/IgoLimsPluginUtils/Tags.java]: Should the interface name be more specific?

Comment (as114) [reply src/main/java/com/velox/sloan/cmo/workflows/IgoLimsPluginUtils/Tags.java]: Tags is the term I use for LIMS tags and Sapio also uses the same term, so I am staying consistent. Anyone working on LIMS would know what it means. Let me know if you still suggest to change the name.

File: src/main/java/com/velox/sloan/cmo/workflows/IgoLimsPluginUtils/AliquotingTags.java (hunk 1 above)
Comment (as114) [reply src/main/java/com/velox/sloan/cmo/workflows/IgoLimsPluginUtils/AliquotingTags.java]: got it.

Comment (as114) [reply src/main/java/com/velox/sloan/cmo/workflows/IgoLimsPluginUtils/AliquotingTags.java]: I applied the changes as suggested above.

**Pull request merged**